packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -191,6 +194,65 @@ export function getDropTargetPosition(
 	];
 }
 
+/**
+ * A Rect hook that takes an array of dragged block client ids and a target client id
+ * and determines if the dragged blocks can be dropped on the target.
+ *
+ * @param {string[]} draggedBlockClientIds The client ids of the dragged blocks.
+ * @param {string}   targetClientId        The client id of the target.
+ * @return {boolean} Whether the dragged blocks can be dropped on the target.
+ */
+export function useIsDropTargetValid( draggedBlockClientIds, targetClientId ) {
```

Comment:
This hook appears to get called even when a drag event isn't happening. Is it possible to move some of the logic to the throttled `useCallback` so that we only figure out the allowed blocks in the throttled function that gets called while a user is dragging? Ideally it'd be good to limit what gets called before the drag actions occur, but it might not be possible to move everything over 🤔

Reply:
This could be refactored to a simple function taking `getBlockType, allowedBlocks, draggedBlockNames, targetBlockName` as parameters and then called inside the throttle.  I'm unsure if determining `isBlockDroppingAllowed` inside the throttle will cause any lag but let's give it a try 😄